lib/Server.js

```diff
@@ -580,17 +581,93 @@ class Server {
   }
 
   showStatus() {
-    const suffix = '/';
-    const uri = `${createDomain(this.options, this.server)}${suffix}`;
+    const useColor = getColorsOption(getCompilerConfigArray(this.compiler));
 
-    const configArr = getCompilerConfigArray(this.compiler);
-    const colors = getColorsOption(configArr);
+    const protocol = this.options.https ? 'https' : 'http';
+    const { hostname, port } = this;
+    const prettyPrintUrl = (newHostname) =>
+      url.format({ protocol, hostname: newHostname, port, pathname: '/' });
+
+    let prettyHostname;
+    let lanUrlForTerminal;
+
+    if (hostname === '0.0.0.0' || hostname === '::') {
+      prettyHostname = 'localhost';
+
+      const localIP = internalIp.v4.sync();
+
+      if (
+        /^10[.]|^172[.](1[6-9]|2[0-9]|3[0-1])[.]|^192[.]168[.]/.test(localIP)
+      ) {
+        // Address is private, format it for later use
```

Comment:
I think this check is not needed, as `0.0.0.0` (`::`) will bind to public address, if available.

Reply:
You can have local public IP (it is very rare edge case, but it is possible), for example I can run it on my server with public IP

Reply:
In that case, I think the public IP should be printed.

Reply:
In this case we print two IP - local and network

Reply:
IMHO, I see no reason not to print the internal IP (the IP address of the NIC) if it's public. It's still a network IP, but just not a LAN.

Reply:
I don't have strong opinion here, it was inspired by CRA and them feedback, but maybe you are right

Reply:
Let's simplify it, we will change it in future after feedback